src/go/rdctl/pkg/lock/lock.go

```diff
@@ -29,12 +34,23 @@ func (lock *BackendLock) Lock(appPaths paths.Paths, action string) error {
 	}
 	// Create an empty file whose presence signifies that the backend is locked.
 	lockPath := filepath.Join(appPaths.AppHome, backendLockName)
-	file, err := os.OpenFile(lockPath, os.O_CREATE|os.O_EXCL, 0o644)
+	file, err := os.OpenFile(lockPath, os.O_CREATE|os.O_RDWR|os.O_EXCL, 0o644)
 	if errors.Is(err, os.ErrExist) {
 		return errors.New("backend lock file already exists; if there is no snapshot operation in progress, you can remove this error with `rdctl snapshot unlock`")
 	} else if err != nil {
 		return fmt.Errorf("unexpected error acquiring backend lock: %w", err)
 	}
+
+	lockData := LockData{
+		Action: action,
+	}
+	encoder := json.NewEncoder(file)
+	encoder.SetIndent("", "  ")
+	if err := encoder.Encode(lockData); err != nil {
+		file.Close()
```

Comment:
I think this should have a `os.Remove(lockPath)` call here. Will change, but waiting for other code review feedback.

Reply:
By "here" you mean if the `encoder.Encode()` call fails? Yes, we should remove it because we can assume it's not processable.